cpp/src/datetime/datetime_ops.cu

```diff
@@ -76,9 +76,30 @@ struct extract_component_operator {
 
     if (time_since_midnight.count() < 0) { time_since_midnight += days(1); }
 
-    auto hrs_  = duration_cast<hours>(time_since_midnight);
-    auto mins_ = duration_cast<minutes>(time_since_midnight - hrs_);
-    auto secs_ = duration_cast<seconds>(time_since_midnight - hrs_ - mins_);
+    auto const hrs_  = [&] { return duration_cast<hours>(time_since_midnight); };
+    auto const mins_ = [&] { return duration_cast<minutes>(time_since_midnight) - hrs_(); };
+    auto const secs_ = [&] {
+      return duration_cast<seconds>(time_since_midnight) - hrs_();
+      -mins_();
+    };
+    auto const millisecs_ = [&] {
+      return duration_cast<milliseconds>(time_since_midnight) - hrs_();
+      -mins_();
+      -secs_();
```

Comment:
We'll need to remove those extra `;`s.

Reply:
I seem to be getting an error if I do so:
```
/nvme/0/pgali/cudf/cpp/src/datetime/datetime_ops.cu(83): error: expected a ";"
          detected during:
            instantiation of "void thrust::cuda_cub::__transform::unary_transform_f<InputIt, OutputIt, thrust::cuda_cub::__transform::no_stencil_tag, TransformOp, Predicate>::operator()(Size) [with InputIt=const cudf::timestamp_D *, OutputIt=int16_t *, TransformOp=cudf::datetime::detail::extract_component_operator<cudf::datetime::detail::datetime_component::YEAR>, Predicate=thrust::cuda_cub::__transform::always_true_predicate, Size=long]" 
/nvme/0/pgali/cudf/cpp/build/_deps/thrust-src/thrust/system/cuda/detail/parallel_for.h(97): here
```
![Screen Shot 2022-10-12 at 1 42 50 PM](https://user-images.githubusercontent.com/11664259/195422813-1b5b4d10-8dc1-45c5-965d-d79291c76b35.png)

Reply:
This should read as follows:
```
auto const secs_ = [&] {
  return duration_cast<seconds>(time_since_midnight) - hrs_() - mins_();
};
```